### ISSUE TITLE
Split gk_field into separate objects

### ISSUE THREAD

**Issue by manauref** [ammarhakim/gkylzero#290]
As I start to add yet another option in apps/gk_field, I am daunted by all the branching. We should split this up into separate objects as it's been suggested before.

**Comment by ammarhakim** [comment]
Yes, I very strongly support this. Please do not use if-statements to implement totally different functionality. Spend time in properly create independent objects, selecting them at initialization time. We need to get into the habit of properly designing code. In general if you have a if-statement that is called every step or in a loop to select between two completely different field solvers, for example, that part of the code needs to be likely refactored. I am glad this is being done.
